src/metrics.ts

```diff
@@ -0,0 +1,99 @@
+import { Registry, Counter, Gauge, collectDefaultMetrics } from 'prom-client'
+
+const METRIC_PREFIX = 'dex_price_estimator_'
+const BUY_AMOUNT_ESTIMATION_METRIC = METRIC_PREFIX + 'buy_amount_estimation'
+const ORDER_BOOK_METRIC = METRIC_PREFIX + 'order_book'
+const ORDER_BOOK_FETCHER_METRIC = METRIC_PREFIX + 'order_book_fetcher'
+
+export const register = new Registry()
+const baseMetricConfig = { registers: [register] }
+const marketsMetricConfig = { ...baseMetricConfig, labelNames: ['base', 'quote', 'hops'] }
+
+// Metrics for get-estimated-buy amount
+export const buyAmountEstimationMetrics = {
+  totalCount: new Counter({
+    name: BUY_AMOUNT_ESTIMATION_METRIC + '_requests_total',
+    help: 'Total requests of the buy amount estimation',
+    ...baseMetricConfig,
+  }),
+
+  count: new Counter({
+    name: BUY_AMOUNT_ESTIMATION_METRIC + '_requests',
+    help: 'Requests of the buy amount estimation by market and hops',
+    ...marketsMetricConfig,
+  }),
+
+  errorsCount: new Counter({
+    name: BUY_AMOUNT_ESTIMATION_METRIC + '_error_total',
+    help: 'Number of failed requests of the buy amount estimation',
+    ...marketsMetricConfig,
+  }),
+
+  durationsTotals: new Gauge({
+    name: BUY_AMOUNT_ESTIMATION_METRIC + '_duration_seconds_total',
+    help: 'Request duration of the buy amount estimation in seconds',
+    ...baseMetricConfig,
+  }),
+
+  durations: new Gauge({
+    name: BUY_AMOUNT_ESTIMATION_METRIC + '_duration_seconds',
+    help: 'Request duration of the buy amount estimation in seconds by market and hops',
+    ...marketsMetricConfig,
+  }),
+}
+
+export const getOrderBookMetrics = {
+  totalCount: new Counter({
+    name: ORDER_BOOK_METRIC + '_requests_total',
+    help: 'Total requests for the order book',
+    ...baseMetricConfig,
+  }),
+
+  count: new Counter({
+    name: ORDER_BOOK_METRIC + '_requests',
+    help: 'Requests for the order book by market and hops',
+    ...marketsMetricConfig,
+  }),
+
+  errorsCount: new Counter({
+    name: ORDER_BOOK_METRIC + '_error_total',
+    help: 'Number of failed requests for the order book',
+    ...marketsMetricConfig,
+  }),
+
+  durationsTotals: new Gauge({
```

Comment:
Is gauge the right metric type for durations, I thought usually people use histograms for that? 
Basically this metric will always report the last query duration and depending on our sampling frequency we could miss requests or read some non-zero values even when there minutes/hours without a new measurement.

We use it in dex-services to measure the times for the different processing stages (orderbook fetching, solution computation, verfication, submission), but it's not really made for that either. It works there because our cycle is very regular (every 5 minutes and it overlays the different stages nicely).

Reply:
I thought about this, you are probably right. I'll re-evaluate tomorrow, and probably do the change

Reply:
Yes, you are right, I think is best to use histogram instead of gauge: 65e730887d7260e6f3205d8a202ccc9a6086c58b